File: v1/nitric-monorepo-turbo/backends/todo/utils/cors.ts
import { HttpMiddleware } from "@nitric/sdk";

export const optionsHandler: HttpMiddleware = (ctx, next) => {
  ctx.res.headers["Content-Type"] = ["text/html; charset=ascii"];
  ctx.res.status = 200;
  ctx.res.body = "OK";

  return ctx;
};

export const cors: HttpMiddleware = (ctx, next) => {
  ctx.res.headers["Access-Control-Allow-Origin"] = ["*"];
  ctx.res.headers["Access-Control-Allow-Headers"] = [
    "Origin, Content-Type, Accept, Authorization",
  ];
  ctx.res.headers["Access-Control-Allow-Methods"] = [
    "GET, PUT, POST, OPTIONS, DELETE",
  ];
  ctx.res.headers["Access-Control-Allow-Credentials"] = ["true"];

  if (next) {
    return next(ctx);
  }

  return ctx;
};

File: v1/flutter/lib/cors.dart
import 'package:nitric_sdk/nitric.dart';

/// Handle Preflight Options requests by returning status 200 to the requests
Future<HttpContext> optionsHandler(HttpContext ctx) async {
  ctx.res.headers["Content-Type"] = ["text/html; charset=ascii"];
  ctx.res.body = "OK";

  return ctx.next();
}

/// Add CORS headers to responses
Future<HttpContext> addCors(HttpContext ctx) async {
  ctx.res.headers["Access-Control-Allow-Origin"] = ["*"];
  ctx.res.headers["Access-Control-Allow-Headers"] = [
    "Origin, X-Requested-With, Content-Type, Accept, Authorization",
  ];
  ctx.res.headers["Access-Control-Allow-Methods"] = [
    "GET, PUT, POST, PATCH, OPTIONS, DELETE",
  ];
  ctx.res.headers["Access-Control-Max-Age"] = ["7200"];

  return ctx.next();
}
